### PR TITLE
feat: Add substrate connection

### PR DESCRIPTION
**Pull request by tcar121293**
Add substrate connection
## Description
Implemented substrate connection with methods:
- NewSubstrateConnection
- GetMetadata
- UpdateMetadata
- GetBlockEvents

## Related Issue Or Context

Closes: #26 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in docs.
- [ ] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: chains/substrate/connection/connection.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+type Connection struct {
+	chain.Chain
+	*rpc.RPC
+	meta        types.Metadata // Latest chain metadata
+	metaLock    sync.RWMutex   // Lock metadata for updates, allows concurrent reads
+	genesisHash types.Hash     // Chain genesis hash
+	name        string         // Chain name
+}
+
+func NewSubstrateConnection(url string, name string) (*Connection, error) {
```
Comment (mpetrun5) [inline chains/substrate/connection/connection.go:26]: Why do we need the name?

**Review by mpetrun5 — COMMENTED** [review]
Some conflicts need to be resolved

**Review by P1sar — APPROVED** [review]
Just fix Matija's comment

File: chains/substrate/connection/connection.go (hunk 1 above)
Comment (tcar121293) [reply chains/substrate/connection/connection.go:26]: Might be needed for some logging stuff... I can remove it for now, and add it later if needed?

**Pull request merged**